mvrx/src/main/kotlin/com/airbnb/mvrx/viewLifecycleAwareLazy.kt

```diff
@@ -0,0 +1,89 @@
+package com.airbnb.mvrx
+
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.RestrictTo
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import java.io.Serializable
+
+/**
+ * This was copied from SynchronizedLazyImpl but modified to automatically initialize in onAttachedToWindow.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY)
+@SuppressWarnings("Detekt.ClassNaming")
+class viewLifecycleAwareLazy<out T>(private val view: View, initializer: (AppCompatActivity, Fragment) -> T) : Lazy<T>, Serializable {
```

Comment:
This is mostly the same as `lifecycleAwareLazy` except it was modified to use window attach state rather than a lifecycle owner.